modules/table/src/test/java/org/apache/ignite/internal/table/distributed/command/PartitionRaftCommandsSerializationTest.java

```diff
@@ -0,0 +1,279 @@
+import org.apache.ignite.internal.schema.Column;
+import org.apache.ignite.internal.schema.NativeTypes;
+import org.apache.ignite.internal.schema.SchemaDescriptor;
+import org.apache.ignite.internal.schema.marshaller.KvMarshaller;
+import org.apache.ignite.internal.schema.marshaller.reflection.ReflectionMarshallerFactory;
+import org.apache.ignite.internal.storage.RowId;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.apache.ignite.internal.tostring.IgniteToStringInclude;
+import org.apache.ignite.internal.tostring.S;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for partition RAFT commands serialization.
+ */
+public class PartitionRaftCommandsSerializationTest extends IgniteAbstractTest {
+    /** Key-value marshaller for tests. */
+    protected static KvMarshaller<TestKey, TestValue> kvMarshaller;
+
+    @BeforeAll
+    static void beforeAll() {
+        var marshallerFactory = new ReflectionMarshallerFactory();
+
+        var schemaDescriptor = new SchemaDescriptor(1, new Column[]{
+                new Column("intKey".toUpperCase(Locale.ROOT), NativeTypes.INT32, false),
+                new Column("strKey".toUpperCase(Locale.ROOT), NativeTypes.STRING, false),
+        }, new Column[]{
+                new Column("intVal".toUpperCase(Locale.ROOT), NativeTypes.INT32, false),
+                new Column("strVal".toUpperCase(Locale.ROOT), NativeTypes.STRING, false),
+        });
+
+        kvMarshaller = marshallerFactory.create(schemaDescriptor, TestKey.class, TestValue.class);
+    }
+
+    @Test
+    public void testUpdateCommand() throws Exception {
+        UpdateCommand cmd = new UpdateCommand(new RowId(1), binaryRow(1), UUID.randomUUID());
+
+        UpdateCommand readCmd = copyCommand(cmd);
+
+        assertEquals(cmd.txId(), readCmd.txId());
+        assertEquals(cmd.getRowId(), readCmd.getRowId());
+        assertArrayEquals(cmd.getRow().bytes(), readCmd.getRow().bytes());
+    }
+
+    @Test
+    public void testRemoveCommand() throws Exception {
+        UpdateCommand cmd = new UpdateCommand(new RowId(1), UUID.randomUUID());
+
+        UpdateCommand readCmd = copyCommand(cmd);
+
+        assertEquals(cmd.txId(), readCmd.txId());
+        assertEquals(cmd.getRowId(), readCmd.getRowId());
+        assertNull(readCmd.getRow());
+    }
+
+    @Test
+    public void testUpdateAllCommand() throws Exception {
+        HashMap<RowId, BinaryRow> rowsToUpdate = new HashMap();
+
+        for (int i = 0; i < 10; i++) {
+            rowsToUpdate.put(new RowId(i), binaryRow(i));
+        }
+
+        var cmd = new UpdateAllCommand(rowsToUpdate, UUID.randomUUID());
+
+        UpdateAllCommand readCmd = copyCommand(cmd);
+
+        assertEquals(cmd.txId(), readCmd.txId());
+
+        for (Map.Entry<RowId, BinaryRow> entry : cmd.getRowsToUpdate().entrySet()) {
+            assertTrue(readCmd.getRowsToUpdate().containsKey(entry.getKey()));
```

Comment:
Let's also check that readCmd has expected number of keys.

Reply:
It is already checked, because all keys are presented in cmd should by presented in readCmd.